src/GridFS/Bucket.php

```diff
@@ -412,6 +413,7 @@ public function getFileIdForStream($stream)
          */
         $typeMap = ['root' => 'stdClass'] + $this->typeMap;
         $file = apply_type_map_to_document($file, $typeMap);
+        assert(is_object($file));
```

Comment:
> I assume this is necessary because psalm only knows that `apply_type_map_to_document()` returns `array|object` and can't infer that the `root` type map will ensure it's a `stdClass` instance. I assume the assertion is enough to silence psalm's complaints.

Yes. The typeMap argument can be defined better (e.g. `array{?root: class-string<T>}`), which should then allow us to use `T` as a return type for the method. I haven't yet played around with it and how it would behave when the given typeMap doesn't contain a `root` element.

> This is the first instance of `assert()` in the library. Is this kosher because of the new handling for PHP 7+ code (see: [Expectations (PHP 7 only)](https://www.php.net/manual/en/function.assert.php#function.assert.expectations))?

Exactly. In development, the `assert` would trip if we're receiving anything other than an object, whilst in a typical production configuration it's entirely optimised away (i.e. doesn't generate an opcode) so has zero performance impact.

I've made a note to double-check the value of `zend.assertions` in our CI processes to ensure we're using development mode.